### PR TITLE
port ColorJitter to prototype transforms

### LINKED ISSUE #5528: Port `transforms.ColorJitter` to `prototype.transforms`
cc @vfdev-5 @datumbox @bjuncek

### PR DESCRIPTION
**Pull request by pmeier**
Closes #5528.

### CODE REVIEW

File: torchvision/prototype/transforms/color.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+from ._utils import is_simple_tensor
+
+
+class ColorJitter(Transform):
+    def __init__(
+        self,
+        brightness: Optional[Union[float, Sequence[float]]] = None,
+        contrast: Optional[Union[float, Sequence[float]]] = None,
+        saturation: Optional[Union[float, Sequence[float]]] = None,
+        hue: Optional[Union[float, Sequence[float]]] = None,
```
Comment (pmeier) [inline torchvision/prototype/transforms/color.py:21]: @FeryET I'm aware you wanted to work on this, but I needed `ColorJitter` for something else. Thus, I went ahead and implemented it. Could you check if the signature is now sufficient for what you were asking in #5646? I also adapted the `_check_input` method to reflect the changes.

File: torchvision/prototype/transforms/color.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            value = [center - value, center + value]
+            if clip_first_on_zero:
+                value[0] = max(value[0], 0.0)
+        elif isinstance(value, collections.abc.Sequence) and len(value) == 2:
+            if not bound[0] <= value[0] <= value[1] <= bound[1]:
+                raise ValueError(f"{name} values should be between {bound}")
+        else:
+            raise TypeError(f"{name} should be a single number or a sequence with length 2.")
+
+        return float(value[0]), float(value[1])
```
Comment (datumbox) [inline torchvision/prototype/transforms/color.py]: You are missing:
https://github.com/pytorch/vision/blob/39772ece7c87ab38b9c2b9df8e7c85e967a739de/torchvision/transforms/transforms.py#L1183-L1186

I understand that you choose to configure `value = None` on the constructor for the same thing, but what you have here is not BC for the case where (0, 0) or (1, 1) etc.

File: torchvision/prototype/transforms/_color.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            if not bound[0] <= value[0] <= value[1] <= bound[1]:
+                raise ValueError(f"{name} values should be between {bound}")
+        else:
+            raise TypeError(f"{name} should be a single number or a sequence with length 2.")
+
+        return float(value[0]), float(value[1])
+
+    # TODO: remove this helper if primitive transforms for F.adjust_{brightness, contrast, saturation, hue} are added
+    class _ColorJitterHelper(Transform):
+        def __init__(self, kernel_tensor: Callable, kernel_pil: Callable, *args: Any, **kwargs: Any) -> None:
```
Comment (datumbox) [inline torchvision/prototype/transforms/_color.py:56]: Can you propose an alternative approach to the nested private class? Here you document quite a few reasons not to use it (logging, instantiation in every call, non-pythonic etc). Should this be a private method for instance?

Also all these weird idioms we are forced to use are papercuts for not having a high-level dispatcher. There are alternatives we could explore that avoid this.

**Review by datumbox — COMMENTED** [review]
Thanks @pmeier, just a couple of comments

File: torchvision/prototype/transforms/color.py (hunk 2 above)
Comment (pmeier) [reply torchvision/prototype/transforms/color.py]: Good catch!

Comment (pmeier) [reply torchvision/prototype/transforms/color.py]: To be fair, the behavior was not BC breaking. The difference is that the transformations would have been performed with a no-op parametrization. So if they don't have a shortcut for that, we get overhead, but not different results.

File: torchvision/prototype/transforms/_color.py (hunk 3 above)
Comment (pmeier) [reply torchvision/prototype/transforms/_color.py]: > Can you propose an alternative approach to the nested private class?

I've added a commit that moves the functionality into a method.

> Also all these weird idioms we are forced to use are papercuts for not having a high-level dispatcher.

Yup.

> There are alternatives we could explore that avoid this.

Do you have anything specific in mind? I've been thinking on this for quite some time, and I haven't found a solution besides having primitive transforms like I mentioned in the code comment or to reinstate the "high level dispatcher kernels".

**Pull request merged**